frb_codegen/src/config/opts_parser.rs

```diff
@@ -87,14 +92,67 @@ pub fn config_parse(mut raw: RawOpts) -> Vec<Opts> {
         );
     }
 
+    // rust/dart shared output path, used
+    let (shared_rust_output_path, shared_dart_output_path) = if raw.rust_input.len() == 1 {
+        // single block case
+        (None, None)
+    } else {
+        // multi-blocks case
+
+        // 1.check all regular path are at the same paths in
+        if !is_same_directory(&rust_output_paths) || !is_same_directory(&dart_output_paths) {
+            panic!("for multi-blocks case, paths in flag `rust-output`/`dart-output` respectively should be in the same directory ");
+        }
+
+        // 2.get proper raw shared rust path
+        let raw_shared_rust_output_path = raw.shared_rust_output.clone().unwrap_or({
+            let p = raw
+                .rust_output
+                .clone()
+                .map(|vec| vec.get(0).cloned().unwrap_or_default())
+                .unwrap_or_default();
+            let directory = Path::new(&p).parent().unwrap_or(Path::new(""));
+
+            let raw_path = raw
+                .shared_rust_output
+                .clone()
+                .unwrap_or_else(|| "./bridge_generated_shares.rs".to_string());
+            let shared_rust_file_name = Path::new(&raw_path).get_file_name(); // erase directory if there is
+            Path::join(directory, shared_rust_file_name)
+                .into_os_string()
+                .into_string()
+                .unwrap()
+        });
+
+        // 3.return rust/dart output path with full directories
+        let shared_rust_output_path =
+            get_valid_canon_paths(&[raw_shared_rust_output_path])[0].clone();
+        let shared_dart_output_path = Path::join(
+            Path::new(&dart_output_paths[0]).parent().unwrap(),
+            format!(
+                "{}.dart",
+                Path::new(&shared_rust_output_path)
+                    .file_name_str()
+                    .unwrap()
+                    .replace(".rs", "")
+                    .replace(".dart", "")
+            ),
+        )
+        .into_os_string()
+        .into_string()
+        .unwrap();
```

Comment:
nit: is it possible for the path resolution logic to be a bit easier? (answer may be no! but I am just worried whether we are over-engineering, or this is just the simplest thing)